subsys/shell/shell_log_backend.c

```diff
@@ -245,7 +249,9 @@ static void put_sync_hexdump(const struct log_backend *const backend,
 	}
 
 	key = irq_lock();
-	shell_cmd_line_erase(shell);
+	if (!flag_cmd_ctx_get(shell)) {
+		shell_cmd_line_erase(shell);
+	}
 	log_output_hexdump(shell->log_backend->log_output, src_level, timestamp,
 			   metadata, data, length, flags);
 	irq_unlock(key);
```

Comment:
Can you also add here
```
if (!flag_cmd_ctx_get(shell)) {
		shell_print_prompt_and_cmd(shell);
	}
```
and remove:
```/* Even though log message is handled notify shell thread which
	 * will print command buffer after the log message.
	 */
	if (IS_ENABLED(CONFIG_MULTITHREADING)) {
		signal = &shell->ctx->signals[SHELL_SIGNAL_LOG_MSG];
		k_poll_signal_raise(signal, 0);
	}
```
i'm not sure how it happened that this functions are different in the way they handle things. They should be the same, except for log_output function.

Reply:
Updated